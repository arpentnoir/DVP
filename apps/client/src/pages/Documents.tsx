import { DataTable, MoreInfo, Text } from '@dvp/vc-ui';
import { Box } from '@mui/material';
import { GridColDef } from '@mui/x-data-grid';
import { useNavigate } from 'react-router-dom';
import { BaseLayout } from '../layouts';

// TODO: Replace with real data
const columns: GridColDef[] = [
  {
    field: 'id',
    headerName: 'ID',
    minWidth: 200,
    flex: 1,
    disableColumnMenu: true,
  },
  {
    field: 'ABN',
    headerName: 'ABN',
    minWidth: 150,
    flex: 1,
    disableColumnMenu: true,
  },
  {
    field: 'businessName',
    headerName: 'Business name',
    minWidth: 150,
    flex: 1,
    disableColumnMenu: true,
  },
  {
    field: 'document',
    headerName: 'Document',
    minWidth: 150,
    flex: 1,
    disableColumnMenu: true,
  },
  {
    field: 'dateIssued',
    headerName: 'Date Issued',
    sortable: false,
    minWidth: 150,
    flex: 1,
    disableColumnMenu: true,
  },
  {
    field: 'moreInfo',
    headerName: '',
    flex: 1,
    renderCell: (params) => (
      <MoreInfo
        params={params}
        items={{ View: () => null, Download: () => null, Revoke: () => null }}
      />
    ),
    disableColumnMenu: true,
    sortable: false,
  },
];

const rows = [
  {
    id: '5940107005302237590',
    document: 'AANZFTA',
    businessName: 'Organisation',
    ABN: '1234 5678 90',
    dateIssued: '01/01/2012',
  },
  {
    id: '5940107005302237591',
    document: 'CITES',
    businessName: 'Organisation',
    ABN: '1234 5678 91',
    dateIssued: '01/01/2012',
  },
  {
    id: '5940107005302237592',
    document: 'AANZFTA',
    businessName: 'Organisation',
    ABN: '1234 5678 92',
    dateIssued: '01/01/2018',
  },
  {
    id: '5940107005302237593',
    document: 'CITES',
    businessName: 'Organisation',
    ABN: '1234 5678 93',
    dateIssued: '01/01/2018',
  },
  {
    id: '5940107005302237594',
    document: 'CITES',
    businessName: 'Organisation',
    ABN: '1234 5678 94',
    dateIssued: '01/01/2032',
  },
  {
    id: '5940107005302237595',
    document: 'AANZFTA',
    businessName: 'Organisation',
    ABN: '1234 5678 95',
    dateIssued: '01/01/2019',
  },
  {
    id: '5940107005302237596',
    document: 'CITES',
    businessName: 'Organisation',
    ABN: '1234 5678 96',
    dateIssued: '01/01/2019',
  },
  {
    id: '5940107005302237597',
    document: 'CITES',
    businessName: 'Organisation',
    ABN: '1234 5678 97',
    dateIssued: '01/01/2020',
  },
  {
    id: '5940107005302237598',
    document: 'AANZFTA',
    businessName: 'Organisation',
    ABN: '1234 5678 98',
    dateIssued: '01/01/2022',
  },
];

export const Documents = () => {
  const navigate = useNavigate();

  const handleIssue = () => {
    navigate('/issue');
  };

  return (
    <BaseLayout title="Documents">
      <Box paddingBottom="40px">
        <Text variant="h1" fontWeight="bold" paddingBottom="24px">
          Documents
        </Text>
        <Text>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam eu
          turpis molestie, dictum est a, mattis tellus. Sed dignissim, metus nec
          fringilla accumsan, risus sem sollicitudin lacus, ut interdum tellus
          elit sed risus. Maecenas eget condimentum velit, sit amet feugiat
          lectus. Class aptent taciti sociosqu ad litora torquent per conubia
          nostra, per inceptos himenaeos. Praesent auctor purus luctus enim
          egestas, ac scelerisque ante pulvinar. Donec ut rhoncus ex.
          Suspendisse ac rhoncus nisl, eu tempor urna. Curabitur vel bibendum
          lorem. Morbi convallis convallis diam sit amet lacinia. Aliquam in
          elementum tellus.
        </Text>
        <Text variant="h3" fontWeight="bold" paddingTop="40px">
          Documents issued
        </Text>
        <Text>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</Text>
      </Box>

      <DataTable
        handleAction={handleIssue}
        toolBarActionLabel="Issue"
        rows={rows}
        columns={columns}
        pageSize={5}
        rowsPerPageOptions={[5]}
        checkboxSelection
      />
    </BaseLayout>
  );
};
