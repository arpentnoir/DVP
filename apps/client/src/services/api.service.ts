import { Auth } from 'aws-amplify';
import axios from 'axios';
import { getApiBaseUrl } from '../config';

// Default config options
const defaultOptions = {
  baseURL: getApiBaseUrl(),
  headers: {
    'Content-Type': 'application/json',
  },
};

export const axiosInstance = axios.create(defaultOptions);

// Set the AUTH token for any request
axiosInstance.interceptors.request.use(async function (config: any) {
  if (!config.headers) {
    config.headers = {};
  }
  try {
    const session = await Auth.currentSession();
    const accessToken = session?.getIdToken().getJwtToken();
    config.headers['Authorization'] = accessToken
      ? `Bearer ${accessToken}`
      : '';
  } catch (_error) {
    // Refresh token, skipping it because we will replace it with SSO and MyGov
  }

  // eslint-disable-next-line @typescript-eslint/no-unsafe-return
  return config;
});
