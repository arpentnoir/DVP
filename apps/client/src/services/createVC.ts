import axios, { AxiosError } from "axios";
import { API_ENDPOINTS, FAIL_CREATE_VC, GENERIC_COO_META_DATA } from "../constants";


export const CreateVC = async (credentialSubject: any) =>{
  try {
    const response = await axios.post(API_ENDPOINTS.ISSUE, {
      credential: {
        credentialSubject: credentialSubject,
        ...GENERIC_COO_META_DATA,
        issuanceDate: new Date().toISOString(),
      },
    });

    return response;
} catch (err: unknown | AxiosError) {
  throw new Error(FAIL_CREATE_VC);
}
};
