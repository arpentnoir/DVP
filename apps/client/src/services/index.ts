export * from './storage.service';
export * from './vc.service';
