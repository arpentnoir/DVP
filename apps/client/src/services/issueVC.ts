import { IssueCredentialRequestSigningMethodEnum } from '@dvp/api-client';
import axios, { AxiosError } from 'axios';
import {
  API_ENDPOINTS,
  FAIL_CREATE_VC,
  GENERIC_OA_META_DATA,
  GENERIC_SVIP_META_DATA,
} from '../constants';

export const IssueVC = async (
  credentialSubject: any,
  credentialType: string
) => {
  try {
    const response = await axios.post(API_ENDPOINTS.ISSUE, {
      credential: {
        ...(credentialType === 'oa'
          ? GENERIC_OA_META_DATA
          : GENERIC_SVIP_META_DATA),
        issuanceDate: new Date().toISOString(),
        credentialSubject: credentialSubject,
      },
      signingMethod:
        credentialType === 'oa'
          ? IssueCredentialRequestSigningMethodEnum.Oa
          : IssueCredentialRequestSigningMethodEnum.Svip,
    });

    return response;
  } catch (err: unknown | AxiosError) {
    throw new Error(FAIL_CREATE_VC);
  }
};
