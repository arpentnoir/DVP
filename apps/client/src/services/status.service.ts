import { IssueCredentialRequestSigningMethodEnum } from '@dvp/api-client';
import { AxiosError } from 'axios';
import { axiosInstance } from './api.service';

import { API_ENDPOINTS, REVOCATION_FAILED_MESSAGE } from '../constants';

enum RevocationType {
  OpenAttestationOcsp = 'OpenAttestationOCSP',
  RevocationList2020Status = 'RevocationList2020Status',
}

enum RevocationStatusCode {
  REVOKED = '1',
  NOT_REVOKED = '0',
}

/**
 * Submits the revocation action to the status queue for processing.
 *
 * @param param0 The data of the VC to be revoked or unrevoked.
 */
export const setRevocationStatus = async ({
  credentialId,
  signingMethod,
  revoke,
}: {
  credentialId: string;
  signingMethod: IssueCredentialRequestSigningMethodEnum;
  revoke: boolean;
}) => {
  try {
    await axiosInstance.post(API_ENDPOINTS.REVOKE, {
      credentialId: `urn:uuid:${credentialId}`,
      credentialStatus: [
        {
          type:
            signingMethod === IssueCredentialRequestSigningMethodEnum.Oa
              ? RevocationType.OpenAttestationOcsp
              : RevocationType.RevocationList2020Status,
          status: revoke
            ? RevocationStatusCode.REVOKED
            : RevocationStatusCode.NOT_REVOKED,
        },
      ],
    });
  } catch (err: unknown | AxiosError) {
    throw new Error(REVOCATION_FAILED_MESSAGE);
  }
};
