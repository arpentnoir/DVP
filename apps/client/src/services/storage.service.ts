import { EncryptedDocument, VerifiableCredential } from '@dvp/api-interfaces';
import { decryptString } from '@govtechsg/oa-encryption';
import { axiosInstance } from './api.service';

import { FAIL_VC_FETCH_DECRYPT_ERR_MSG } from '../constants';

export const getVC = async (storageUrl: string) => {
  return axiosInstance.get<{ document: EncryptedDocument }>(storageUrl);
};

/**
 * 1. Extracts the VC storage endpoint and decryption key
 * 2. Fetches encrypted VC
 * 3. Decrypts VC
 * 4. Returns JSON parsed VC
 *
 * @param params the query string that contains payload for the vc storage endpoint and decryption key
 * @returns document
 */
export const fetchAndDecryptVC = async (params: string) => {
  try {
    const searchParams = new URLSearchParams(params);

    const anchorStr: string | null = searchParams.get('q');

    const anchor = anchorStr ? JSON.parse(anchorStr) : {};

    const storageUrl = anchor.payload.uri as string;

    const key = anchor.payload.key as string;

    const res = await getVC(storageUrl);

    const document = decryptString({ ...res.data.document, key });

    return JSON.parse(document) as VerifiableCredential;
  } catch (err) {
    throw new Error(FAIL_VC_FETCH_DECRYPT_ERR_MSG);
  }
};
