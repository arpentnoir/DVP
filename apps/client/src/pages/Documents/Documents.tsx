/**
* @module Documents
* @description This module exports the Documents page component, which displays a
  list of Verifiable Credentials and their details that were issued 
  by the associated ABN of the user, along with options to search,
  sort, and navigate to the issue page.
*/
import { CredentialsResponseItem } from '@dvp/api-client';
import { DataTable, MoreInfo, Text } from '@dvp/vc-ui';
import { Box } from '@mui/material';
import {
  GridColDef,
  GridRenderCellParams,
  GridValueFormatterParams,
} from '@mui/x-data-grid';
import { useNavigate } from 'react-router-dom';
import { useQueryApi } from '../../hooks/useQueryApi';
import { BaseLayout } from '../../layouts';
import { getVerifiableCredentials } from '../../services';

/**
 * @typedef GridColType
 * @type {object}
 * @property {string} field - The unique identifier for the column.
 * @property {string} headerName - The column's display name in the header.
 * @property {number} flex - The column's flexible width ratio, relative to other columns.
 * @property {boolean} disableColumnMenu - Indicates whether the column menu should be disabled or not.
 * @property {boolean} sortable - Indicates whether the column is sortable or not.
 * @property {number} [minWidth] - The minimum width of the column, if defined.
 * @property {(params: GridRenderCellParams) => JSX.Element} [renderCell] - An optional render function for the column's cells.
 * @property {(params: GridValueFormatterParams) => string} [valueFormatter] - An optional formatter function for the column's cells.

 */
interface GridColType {
  field: string;
  headerName: string;
  flex: number;
  disableColumnMenu: boolean;
  sortable: boolean;
  minWidth?: number;
  renderCell?: (params: GridRenderCellParams) => JSX.Element;
  valueFormatter?: (params: GridValueFormatterParams) => string;
}

/**
 * @type {GridColDef<GridColType>[]}
 * @description The columns configuration for the data table.
 */
const columns: GridColDef<GridColType>[] = [
  {
    field: 'documentNumber',
    headerName: 'Document ID',
    minWidth: 150,
    flex: 1,
    disableColumnMenu: true,
    sortable: false,
  },
  {
    field: 'importerName',
    headerName: 'Business Name',
    minWidth: 200,
    flex: 1,
    disableColumnMenu: true,
    sortable: false,
  },
  {
    field: 'exporterOrManufacturerAbn',
    headerName: 'ABN',
    minWidth: 210,
    flex: 1,
    disableColumnMenu: true,
    sortable: false,
  },
  {
    field: 'freeTradeAgreement',
    headerName: 'Document Type',
    minWidth: 180,
    flex: 1,
    disableColumnMenu: true,
    sortable: false,
  },
  {
    field: 'issueDate',
    headerName: 'Issue Date',
    minWidth: 260,
    flex: 1,
    disableColumnMenu: true,
    sortable: true,
  },
  {
    field: 'createdBy',
    headerName: 'Issuer ID',
    minWidth: 150,
    flex: 1,
    disableColumnMenu: true,
    sortable: false,
  },
  {
    field: 'isRevoked',
    headerName: 'Document Status',
    minWidth: 170,
    flex: 1,
    disableColumnMenu: true,
    sortable: false,
    valueFormatter: (params: GridValueFormatterParams<boolean>) =>
      params.value ? 'Revoked' : 'Current',
  },
  {
    field: 'moreInfo',
    headerName: '',
    flex: 1,
    renderCell: (params) => (
      <MoreInfo
        params={params}
        items={{ View: () => null, Download: () => null, Revoke: () => null }}
      />
    ),
    disableColumnMenu: true,
    sortable: false,
  },
];

/**
 * @const {string} errorMessage
 * @description The error message to be displayed when an error occurs while fetching the Verifiable Credentials.
 */
export const errorMessage =
  'An error occurred while fetching the Verifiable Credentials';

/**
* @function Documents
* @description A component that displays a list of Verifiable Credentials and their details
  that were issued by the associated ABN of the user, along with options to search,
  sort, and navigate to the issue page.
* @returns {JSX.Element} The rendered Documents component.
*/
export const Documents = () => {
  const navigate = useNavigate();

  const handleIssue = () => {
    navigate('/issue');
  };

  const { handleSearch, handleSort, paginationControls, state } =
    useQueryApi<CredentialsResponseItem>(getVerifiableCredentials, {
      errorMessage,
    });

  return (
    <BaseLayout title="Documents">
      <Box paddingBottom="40px">
        <Text variant="h1" fontWeight="bold" paddingBottom="24px">
          Documents
        </Text>
        <Text>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam eu
          turpis molestie, dictum est a, mattis tellus. Sed dignissim, metus nec
          fringilla accumsan, risus sem sollicitudin lacus, ut interdum tellus
          elit sed risus. Maecenas eget condimentum velit, sit amet feugiat
          lectus. Class aptent taciti sociosqu ad litora torquent per conubia
          nostra, per inceptos himenaeos. Praesent auctor purus luctus enim
          egestas, ac scelerisque ante pulvinar. Donec ut rhoncus ex.
          Suspendisse ac rhoncus nisl, eu tempor urna. Curabitur vel bibendum
          lorem. Morbi convallis convallis diam sit amet lacinia. Aliquam in
          elementum tellus.
        </Text>
        <Text variant="h3" fontWeight="bold" paddingTop="40px">
          Documents issued
        </Text>
        <Text>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</Text>
      </Box>

      <DataTable
        handleAction={handleIssue}
        toolBarActionLabel="Issue"
        handleOnSort={handleSort}
        handleOnSearch={handleSearch}
        columns={columns}
        rows={state.data ? state.data : []}
        paginationControls={paginationControls}
        isLoading={state.isLoading}
        error={state.errorMessage}
        checkboxSelection
      />
    </BaseLayout>
  );
};
