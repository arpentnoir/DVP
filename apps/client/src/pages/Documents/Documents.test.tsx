import { render } from '@dvp/vc-ui';
import { fireEvent, waitFor } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { useQueryApi } from '../../hooks';
import { getVerifiableCredentials } from '../../services';
import { Documents, errorMessage } from './Documents';

jest.mock('../../hooks/useQueryApi/useQueryApi', () => {
  return {
    useQueryApi: jest.fn(() => ({
      handleSearch: () => null,
      handleSort: () => null,
      paginationControls: {},
      state: { rows: [] },
    })),
  };
});

describe('Documents', () => {
  it('should render correctly', () => {
    const { baseElement } = render(<Documents />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should pass the error message specified in the documents component into the useQueryApi hook', () => {
    render(<Documents />);

    expect(useQueryApi as jest.Mock).toBeCalledWith(getVerifiableCredentials, {
      errorMessage,
    });
  });

  it('should navigate to the issue page when table action button is clicked', async () => {
    const history = createMemoryHistory();

    const { getByTestId } = render(<Documents />, { navigator: history });

    await waitFor(() => {
      expect(history.location.pathname).not.toBe('/issue');
    });

    fireEvent.click(getByTestId('button:Issue'));

    await waitFor(() => {
      expect(history.location.pathname).toBe('/issue');
    });
  });
});
