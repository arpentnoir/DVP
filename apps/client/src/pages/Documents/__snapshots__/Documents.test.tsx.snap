// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documents should render correctly 1`] = `
<body>
  <div>
    <div
      class="MuiContainer-root MuiContainer-disableGutters css-1y9y4eh-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-m31y3a"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-1ne5dp9-MuiTypography-root"
          tabindex="0"
        >
          Documents
        </h1>
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          tabindex="0"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam eu turpis molestie, dictum est a, mattis tellus. Sed dignissim, metus nec fringilla accumsan, risus sem sollicitudin lacus, ut interdum tellus elit sed risus. Maecenas eget condimentum velit, sit amet feugiat lectus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Praesent auctor purus luctus enim egestas, ac scelerisque ante pulvinar. Donec ut rhoncus ex. Suspendisse ac rhoncus nisl, eu tempor urna. Curabitur vel bibendum lorem. Morbi convallis convallis diam sit amet lacinia. Aliquam in elementum tellus.
        </p>
        <h3
          class="MuiTypography-root MuiTypography-h3 css-12hh03f-MuiTypography-root"
          tabindex="0"
        >
          Documents issued
        </h3>
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          tabindex="0"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </p>
      </div>
      <div
        class="MuiBox-root css-0"
        style="height: 600px; width: 100%;"
      >
        <div
          aria-colcount="9"
          aria-multiselectable="false"
          aria-rowcount="1"
          class="MuiDataGrid-root MuiDataGrid-root--densityStandard MuiDataGrid-withBorderColor css-1usso8h-MuiDataGrid-root"
          role="grid"
        >
          <div>
            <div
              class="css-qz6wf1-MuiStack-root"
            >
              <div
                aria-label="Search"
                class="MuiFormControl-root MuiTextField-root css-3be3ve-MuiFormControl-root-MuiTextField-root-MuiDataGrid-toolbarQuickFilter"
              >
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1ptx2yq-MuiInputBase-root-MuiInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch MuiInputBase-inputAdornedEnd css-c63i49-MuiInputBase-input-MuiInput-input"
                    id=":r0:"
                    placeholder="Searchâ€¦"
                    type="search"
                    value=""
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                  />
                </div>
              </div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium css-ndac8y-MuiButtonBase-root-MuiButton-root"
                data-testid="button:Issue"
                style="margin-left: auto;"
                tabindex="0"
                type="button"
              >
                <div
                  class="css-doi3ij-MuiStack-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="AddIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                    />
                  </svg>
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                    data-testid="button:Issue:label"
                    tabindex="0"
                  >
                    Issue
                  </p>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiDataGrid-main css-204u17-MuiDataGrid-main"
          >
            <div
              class="MuiDataGrid-columnHeaders MuiDataGrid-withBorderColor css-1iyq7zh-MuiDataGrid-columnHeaders"
              style="min-height: 56px; max-height: 56px; line-height: 56px;"
            >
              <div
                class="MuiDataGrid-columnHeadersInner MuiDataGrid-columnHeadersInner--scrollable css-gl260s-MuiDataGrid-columnHeadersInner"
                role="rowgroup"
                style="transform: translate3d(0px, 0px, 0px);"
              >
                <div
                  aria-rowindex="1"
                  class="css-yrdy0g-MuiDataGrid-columnHeaderRow"
                  role="row"
                >
                  <div
                    aria-colindex="1"
                    aria-sort="none"
                    class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--alignCenter MuiDataGrid-withBorderColor MuiDataGrid-columnHeaderCheckbox"
                    data-field="__check__"
                    role="columnheader"
                    style="height: 56px; width: 50px; min-width: 50px; max-width: 50px;"
                    tabindex="0"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderDraggableContainer"
                      draggable="false"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainer"
                      >
                        <div
                          class="MuiDataGrid-columnHeaderTitleContainerContent"
                        >
                          <span
                            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary MuiDataGrid-checkboxInput css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                          >
                            <input
                              aria-label="Select all rows"
                              class="PrivateSwitchBase-input css-1m9pwf3"
                              data-indeterminate="false"
                              tabindex="-1"
                              type="checkbox"
                            />
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                              data-testid="CheckBoxOutlineBlankIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                              />
                            </svg>
                            <span
                              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                            />
                          </span>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                      style="min-height: 56px; opacity: 1;"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                        data-testid="SeparatorIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11 19V5h2v14z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    aria-colindex="2"
                    aria-label="Document ID"
                    aria-sort="none"
                    class="MuiDataGrid-columnHeader MuiDataGrid-withBorderColor"
                    data-field="documentNumber"
                    role="columnheader"
                    style="height: 56px; width: 0px; min-width: 0; max-width: 0;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderDraggableContainer"
                      draggable="false"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainer"
                      >
                        <div
                          class="MuiDataGrid-columnHeaderTitleContainerContent"
                        >
                          <div
                            aria-label=""
                            class="MuiDataGrid-columnHeaderTitle css-t89xny-MuiDataGrid-columnHeaderTitle"
                            data-mui-internal-clone-element="true"
                          >
                            Document ID
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                      style="min-height: 56px; opacity: 1;"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                        data-testid="SeparatorIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11 19V5h2v14z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    aria-colindex="3"
                    aria-label="Business Name"
                    aria-sort="none"
                    class="MuiDataGrid-columnHeader MuiDataGrid-withBorderColor"
                    data-field="importerName"
                    role="columnheader"
                    style="height: 56px; width: 0px; min-width: 0; max-width: 0;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderDraggableContainer"
                      draggable="false"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainer"
                      >
                        <div
                          class="MuiDataGrid-columnHeaderTitleContainerContent"
                        >
                          <div
                            aria-label=""
                            class="MuiDataGrid-columnHeaderTitle css-t89xny-MuiDataGrid-columnHeaderTitle"
                            data-mui-internal-clone-element="true"
                          >
                            Business Name
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                      style="min-height: 56px; opacity: 1;"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                        data-testid="SeparatorIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11 19V5h2v14z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              style="flex: 1 1 0px; overflow: auto;"
            >
              <div
                class="MuiDataGrid-virtualScroller css-axafay-MuiDataGrid-virtualScroller"
                style="overflow-x: hidden;"
              >
                <div
                  class="MuiDataGrid-overlayWrapper css-ql19fo-MuiDataGrid-overlayWrapper"
                >
                  <div
                    class="MuiDataGrid-overlayWrapperInner css-1akuw9y-MuiDataGrid-overlayWrapperInner"
                    style="height: -56px; width: 0px;"
                  >
                    <div
                      class="MuiDataGrid-overlay css-1w53k9d-MuiDataGrid-overlay"
                    >
                      No rows
                    </div>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-virtualScrollerContent MuiDataGrid-virtualScrollerContent--overflowed css-1kwdphh-MuiDataGrid-virtualScrollerContent"
                  style="width: auto; height: 1px; min-height: auto;"
                >
                  <div
                    class="MuiDataGrid-virtualScrollerRenderZone css-s1v7zr-MuiDataGrid-virtualScrollerRenderZone"
                    style="transform: translate3d(0px, undefinedpx, 0px);"
                  />
                </div>
              </div>
            </div>
            <div
              class="Mui-resizeTriggers"
            >
              <div
                class="expand-trigger"
              >
                <div
                  style="width: 1px; height: 1px;"
                />
              </div>
              <div
                class="contract-trigger"
              />
            </div>
          </div>
          <div
            class="MuiDataGrid-footerContainer MuiDataGrid-withBorderColor css-wop1k0-MuiDataGrid-footerContainer"
          >
            <div />
            <div
              class="MuiTablePagination-root css-1gl81ut-MuiTablePagination-root"
            >
              <div
                class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
              >
                <div
                  class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
                />
                <p
                  class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
                >
                  1â€“1 of 1
                </p>
                <div
                  class="MuiTablePagination-actions"
                >
                  <button
                    aria-label="Go to previous page"
                    class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                    data-testid="button:cursor-pagination-prev"
                    disabled=""
                    tabindex="-1"
                    title="Go to previous page"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="KeyboardArrowLeftIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                      />
                    </svg>
                  </button>
                  <button
                    aria-label="Go to next page"
                    class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                    data-testid="button:cursor-pagination-next"
                    disabled=""
                    tabindex="-1"
                    title="Go to next page"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="KeyboardArrowRightIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
